app/models/customer.py

```diff
@@ -37,10 +37,10 @@ def __repr__(self):
 class Email(db.Model):
     __tablename__ = "email"
     id: Mapped[int] = mapped_column(Integer, primary_key=True)
-    email: Mapped[str] = mapped_column(String(255))
+    email: Mapped[Optional[str]] = mapped_column(String(255))
     customer_id: Mapped[int] = mapped_column(ForeignKey("customer.id", ondelete="CASCADE"))
 
-    customer: Mapped["Customer"] = relationship("Customer", back_populates="emails")
+    customer: Mapped[Optional["Customer"]] = relationship("Customer", back_populates="emails")
```

Comment:
I think we need to require at least one email address.  What do you think?

Reply:
It still can't be null on the Customer side of the relationship. Now that you point that out though, having the email nullable in the Email model is no longer necessary. When I was originally writing this I was removing the email from the session first and then deleting it, and doing it that way it had to be nullable in the Emailmodel. Pushing another commit now to remove the optional from the models and rebasing.